Ninjathulhu/Ninjathulhu.Game.Test/EntityComponentsTests.cs

```diff
@@ -79,5 +79,13 @@ public void removing_a_component_that_has_not_been_registered_does_not_throw()
 
             act.ShouldNotThrow();
         }
+
+        [Test]
+        public void trying_to_attach_a_non_Component_to_an_entity_should_throw()
```

Comment:
Figure we should verify that this actually does throw, and having a test kind of documents that we mean it this way :)